server/src/handler_controller.c

```diff
@@ -246,68 +268,88 @@ GString *grant_route(const char *train_id, const char *source_id, const char *de
+		              train_id, source_id, destination_id);
+	} else if (strcmp(g_route_id_copy->str, "not_grantable") == 0) {
+		syslog_server(LOG_WARNING, 
+		              "Grant route - train: %s from: %s to: %s - conflicting routes in use",
+		              train_id, source_id, destination_id);
+	} else if (strcmp(g_route_id_copy->str, "not_clear") == 0) {
+		syslog_server(LOG_WARNING, 
+		              "Grant route - train: %s from: %s to: %s - route blocked or source signal not stop",
+		              train_id, source_id, destination_id);
+	} else {
+		syslog_server(LOG_WARNING, 
+		              "Grant route - train: %s from: %s to: %s - route could not be granted (message: %s)",
+		              train_id, source_id, destination_id, g_route_id_copy->str);
+	}
+	return g_route_id_copy;
 }
 
 const char *grant_route_id(const char *train_id, const char *route_id) {
+	if (train_id == NULL || route_id == NULL) {
+		syslog_server(LOG_ERR, "Grant route id - invalid (NULL) parameter(s)");
+		return "not_grantable";
+	}
 	pthread_mutex_lock(&interlocker_mutex);
-
 	// Check whether the route can be granted
 	t_interlocking_route * const route = get_route(route_id);
 	GArray * const granted_conflicts = get_granted_route_conflicts(route_id);
 	if (granted_conflicts == NULL) {
-		syslog_server(LOG_ERR, "grant_route_id: granted_conflicts is NULL");
 		pthread_mutex_unlock(&interlocker_mutex);
+		syslog_server(LOG_WARNING, 
+		              "Grant route id - train: %s route: %s - granted_conflicts is NULL",
+		              train_id, route_id);
 		return "not_grantable";
 	}
 	const bool hasGrantedConflicts = (granted_conflicts->len > 0);
 	g_array_free(granted_conflicts, true);
 	if (route->train != NULL || hasGrantedConflicts) {
 		pthread_mutex_unlock(&interlocker_mutex);
+		syslog_server(LOG_WARNING, 
+		              "Grant route id - train: %s route: %s - route already granted "
+		              "or conflicts with granted routes",
+		              train_id, route_id);
 		return "not_grantable";
 	}
 
 	// Check whether the route is physically available
 	if (!get_route_is_clear(route_id)) {
 		pthread_mutex_unlock(&interlocker_mutex);
+		syslog_server(LOG_WARNING, 
+		              "Grant route id - train: %s route: %s - route is not clear",
+		              train_id, route_id);
 		return "not_clear";
 	}
 
 	// Grant the route to the train and mark it unavailable
 	route->train = strdup(train_id);
 	
 	if (route->train == NULL) {
-		syslog_server(LOG_ERR, "grant_route_id: Unable to allocate memory for route->train");
 		pthread_mutex_unlock(&interlocker_mutex);
+		syslog_server(LOG_ERR, 
+		              "Grant route id - train: %s route: %s - unable to allocate memory for route->train",
+		              train_id, route_id);
 		return "not_grantable";
 	}
+	
+	syslog_server(LOG_INFO, 
+	              "Grant route id - train: %s route: %s - granting in progress, setting points and signals", 
```

Comment:
IMO up until line 335/336 we've been checking whether the route may be granted. The comment on line 335 also fits with that - there we start to "grant" the route.   
Also, "granting in progress" does not have to be at the start of the actual granting; then it would say "starting to grant..." or something similar.